jina/proto/jina.proto

```diff
@@ -281,68 +224,74 @@ message RelatedEntity {
     optional uint32 shard_id = 4; // the id of the shard it belongs to, if it is a shard
 }
 
+
 /**
- * Represents a Request
+ * Represents a ControlRequestProto
  */
-message RequestProto {
+message ControlRequestProto {
 
-    string request_id = 1; // the unique ID of this request. Multiple requests with the same ID will be gathered
+    HeaderProto header = 1; // header contains meta info defined by the user
 
-    oneof body {
-        ControlRequestProto control = 2; // a control request
-        DataRequestProto data = 3; // a data request
+    enum Command {
+            STATUS = 0; // check the status of the BasePod
+            ACTIVATE = 1; // used to add Peas to a Pod
+            DEACTIVATE = 2; // used to remove Peas from a Pod
     }
 
-    /**
-     * Represents a general data request
-     */
-    message DataRequestProto {
-        repeated DocumentProto docs = 1; // a list of Documents to query
-        repeated DocumentProto groundtruths = 2; // a list of groundtruth Document you want to evaluate it with
-    }
+    Command command = 2; // the control command
+
+    repeated RelatedEntity relatedEntities = 3; // list of entities this ControlMessage is related to
+}
 
-    /**
-     * Represents a control request used to control the BasePod
-     */
-    message ControlRequestProto {
-        enum Command {
-            TERMINATE = 0; // shutdown the BasePod
-            STATUS = 1; // check the status of the BasePod
-            ACTIVATE = 2; // used to add Peas to a Pod
-            DEACTIVATE = 3; // used to remove Peas from a Pod
-        }
-        Command command = 1; // the control command
-
-        repeated RelatedEntity relatedEntities = 2; // list of entities this ControlMessage is related to
-    }
 
-    HeaderProto header = 4; // header contains meta info defined by the user
+/**
+ * Represents a DataRequestProto
+ */
+message DataRequestProto {
 
-    google.protobuf.Struct parameters = 5; // extra kwargs that will be used in executor
+    HeaderProto header = 1; // header contains meta info defined by the user
 
-    repeated RouteProto routes = 6; // status info on every routes
+    google.protobuf.Struct parameters = 2; // extra kwargs that will be used in executor
 
-    StatusProto status = 7; // status info
+    repeated RouteProto routes = 3; // status info on every routes
 
+    repeated DocumentProto docs = 4; // the docs in this request
```

Comment:
May we want to wrap `docs` and `groundtruths` together inside `data` as it used to?

Reply:
hmm I have no strong opinion here, but my impression is that most people are rather interested in the docs and only a small minority in groundtruth.
Anyway I added a small hack for backwards compatibility. Trying to access `data` now will return a named tuple wrapping `docs` and `groundtruths` 
Its tested here https://github.com/jina-ai/jina/pull/4051/files#diff-a1b83d7177090e17771956ad8ec7a3c5cac4c9cc795a6e65e16a8d96b32c1f4fR81